README.md

```diff
@@ -1,2 +1,859 @@
+
+- Required or Application Cookies
+- Personalization Cookies
+- Statistics Cookies
+- Advertising and Marketing Cookies
+
+For more information please visit [XcooBee Cookie Classification](https://www.xcoobee.com/docs/xcoobee-concepts/cookies/).
+
+
+
+## Example Applications
+
+We have included two example applications that make use of the XCK so you can check it in action yourself.
+
+### Single Page Application
+
+Our cafe application demonstrates the use of XCK in a single page application. You can unzip and deploy on a webserver like IIS or Apache. The files are located in `examples/cafe-spa.zip`. Review the index.html file for instructions/explanations.
+
+### Sample Request/Response app based on NodeJs & Express
+
+Our pizza application demonstrates the use of the XCK in a request/response based application. You can unzip and deploy it directly on your system. You will need nodejs installed on your system. Review the `README.md` file for instructions/explanations.
+ 
+## How does this Work
+
+### The Cookie
+
+![the cookie](cookie.png)
+
+The XCK displays a floating cookie icon in the corner of the screen you select. By clicking on the cookie users open a popup consent dialog that allows them to interact with your site.
+
+![showing XCK cookie popup](cookie_popup.png)
+
+The XCK can automatically determine when a full consent dialog or just the cookie is needed. Thus, making the site visit more pleasant and frictionless.
+
+When using the XCK you:
+
+- a) first determine which cookies you are currently using on your site
+- b) then select which XcooBee type (classification) best fits each cookie
+- c) display to your site visitors the XCK consent dialog pop-up to ask for permission
+- d) after obtaining consent for a type of cookie, set the cookies in that group for your user using either a handler or target url pattern 
+
+After the cookie consent is already obtained, the XCK will not display another popup. You can query the status via JavaScript to set the cookies again, or track this on the backend side if you using Request/Response type application.
+
+
+
+### Example walk through
+
+Let's assume an example of a JSP driven shopping site that uses four cookies:
+
+`a` You determine that you use the following four cookies on your site.
+
+1. Your JSP session cookie
+2. A user cookie to keep track of user's theme
+3. A site statistics cookie from your webserver
+4. A global statistics cookie from Google Analytics
+
+`b` You classify the cookies into the following types
+
+Application: JSP Session Cookie
+
+Personalization: user cookie
+
+Statistics: local webserver cookie & google analytics cookie
+
+`c` You [initialize](#c-load-xcoobee-cookie-kit) display the XCK script to displays user consent popup. 
+
+`d` User makes selection and clicks OK.
+
+You set the needed cookies using a JavaScript handler process that gets invoked by the XCK.
+
+
+### Cookie Pulses
```

Comment:
I am concerned about the use of non-semantic names in the code for the four different consent states. If we later change our minds that we want to style the cookie differently when the consent is known (currently styled as a green cookie), then the non-semantic name of `green` will conflict with the updated style (assuming it is no longer green). If we use something like "consent known" instead of "green", then we have the freedom to style that state however we want.

One case I can think of where we may need to change the styling is due to accessibility reasons. If my understanding is correct, I believe color blind people have a hard time or can't distinguish between green and red. So, they can't rely on the color they see to know the consent state.